src/Player.ts

```diff
@@ -264,14 +286,197 @@ export class Player implements ISerializable<SerializedPlayer> {
     return this.terraformRating = value;
   }
 
-  public getProduction(resource: Resources): number {
-    if (resource === Resources.MEGACREDITS) return this.megaCreditProduction;
-    if (resource === Resources.STEEL) return this.steelProduction;
-    if (resource === Resources.TITANIUM) return this.titaniumProduction;
-    if (resource === Resources.PLANTS) return this.plantProduction;
-    if (resource === Resources.ENERGY) return this.energyProduction;
-    if (resource === Resources.HEAT) return this.heatProduction;
-    throw new Error('Resource ' + resource + ' not found');
```

Comment:
One reason I think this change is great is a point that was mentioned earlier. With this in place it is must simpler to see places where we may increase a resource. We also no longer need to check here for when a resource is not found. The resources for a `Player` are static. This mapping we used to always perform would also run 6 times slower. Updating `heatProduction` would require first comparing `resource` to `MEGACREDITS`, `STEEL`, `TITANIUM`, `PLANTS`, and `ENERGY` when we know that we need to update heat production.